weld/subsystem/src/main/java/org/jboss/as/weld/WeldStartCompletionService.java

```diff
@@ -44,13 +48,27 @@
     private final InjectedValue<WeldBootstrapService> bootstrap = new InjectedValue<WeldBootstrapService>();
 
     private final ClassLoader classLoader;
+    private final List<ServiceController> serviceControllers;
 
-    public WeldStartCompletionService(ClassLoader classLoader) {
+    public WeldStartCompletionService(ClassLoader classLoader, List<ServiceController> serviceControllers) {
         this.classLoader = classLoader;
+        this.serviceControllers = serviceControllers;
     }
 
     @Override
     public void start(final StartContext context) throws StartException {
+        StabilityMonitor monitor = new StabilityMonitor();
+        for (ServiceController controller : serviceControllers) {
+            monitor.addController(controller);
+        }
+        try {
+            monitor.awaitStability();
+        } catch (InterruptedException ex) {
+            // no-op
```

Comment:
Hm, maybe something like this would be more appropriate:
```
} catch (InterruptedException ex) {
   Thread.currentThread().interrupt();
   throw new RuntimeException(ex);
}
```